src/lib/statistics/abstract_column_statistics.hpp

```diff
@@ -0,0 +1,84 @@
+#pragma once
+
+#include <memory>
+
+#include "all_type_variant.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+class AbstractColumnStatistics;
+
+struct ColumnValueEstimate {
+  float selectivity{0.0f};
+  std::shared_ptr<AbstractColumnStatistics> column_statistics;
+};
+
+struct ColumnColumnEstimate {
+  float selectivity{0.0f};
+  std::shared_ptr<AbstractColumnStatistics> left_column_statistics;
+  std::shared_ptr<AbstractColumnStatistics> right_column_statistics;
+};
+
+class AbstractColumnStatistics {
```

Comment:
Abstract or Base?

Reply:
Can live with both, like Abstract better because the class mostly defines the interface of the deriving classes, that's its main purpose.

Reply:
We are using "Base" if we only have that class because we need a non-templated super class. Only if we have multiple implementations, it's an "Abstract"